csharp_dotnetcore/17.Proactive-Messages/State/JobLog.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.BotBuilderSamples
+{
+    /// <summary>Contains a dictionary of job data, indexed by job number.</summary>
+    public class JobLog : Dictionary<long, JobLog.JobData>
+    {
```

Comment:
I've never seen this pattern before quite like this. It's very interesting, and pretty clear to read.